src/main/java/zbot/StorageManager.java

```diff
@@ -0,0 +1,145 @@
+     * If a task is marked as done (indicated by a "1" in the file), it is marked as completed when created.
+     *
+     * @return a list of tasks loaded from the file
+     * @throws ZBOTFileNotFoundException if the file cannot be found
+     * @throws IOException if the file format is invalid or there is an error reading the file
+     */
+    public List<Task> loadExistingFile() throws ZBOTFileNotFoundException, IOException {
+        List<Task> ans = new ArrayList<>();
+        try {
+            File f = new File(filePath);
+            Scanner s = new Scanner(f);
+            while (s.hasNext()) {
+                String curr = s.nextLine();
+                String[] parts = curr.split(" \\| ");
+                switch (parts[0]) {
+                case "T" :
+                    ToDoTask a = new ToDoTask(parts[2]);
+                    if (parts[1].equals("1")) {
+                        a.markDone();
+                    }
+                    ans.add(a);
+                    break;
+                case "D":
+                    DeadlineTask b = new DeadlineTask(parts[2], parts[3]);
+                    if (parts[1].equals("1")) {
+                        b.markDone();
+                    }
+                    ans.add(b);
+                    break;
+                case "E" :
+                    EventTask c = new EventTask(parts[2], parts[3], parts[4]);
+                    if (parts[1].equals("1")) {
+                        c.markDone();
+                    }
+                    ans.add(c);
+                    break;
+                default :
+                    throw new IOException("File format is invalid");
+                }
+            }
+        } catch (FileNotFoundException e) {
+            throw new ZBOTFileNotFoundException("The data could not be found");
+        }
+        return ans;
+    }
+
+    /**
+     * Saves the task list to a file.
+     * <p>
+     * This method saves all tasks in the given task list to a file. Each task is serialized in a specific format:
+     * - "T" for ToDoTask
+     * - "D" for DeadlineTask
+     * - "E" for EventTask
+     * The file will include the task's done status (1 for done, 0 for not done), description, and for specific task types, additional details such as deadline or event dates.
+     * <p>
+     * If the directory where the file is to be stored does not exist, it will be created.
+     *
+     * @param taskList the task list containing tasks to be saved
+     * @throws IOException if an error occurs while creating directories or writing to the file
+     */
+    public void saveToFile(TaskList taskList) throws IOException {
+        File file = new File(filePath);
+        File parentDir = file.getParentFile();
+        if (parentDir != null && !parentDir.exists()) {
+            if (!parentDir.mkdirs()) {
+                throw new IOException("Sorry !! Failed to create directory: " + parentDir.getAbsolutePath());
+            }
+        }
+        try {
```

Comment:
consider making this a try-with-resources, so that the `FileWriter` is automatically closed for you. right now, you're only closing the `FileWriter` within the main `try` block, so if an exception is thrown, you're not actually closing it. this could also be done with a `finally` block.

```suggestion
        try (FileWriter fileWriter = new FileWriter(filePath)) {
```

Reply:
thanks for the input! implemented it:)